### PR TITLE
Removed Navigator from TermsOfServiceFragment, SurveySelectorFragment and OfflineAreasFragment

### LINKED ISSUE #2776: MainActivity.onNavigate java.lang.IllegalArgumentException
```console
 Fatal Exception: java.lang.IllegalArgumentException: Navigation action/destination org.openforis.ground:id/action_home_screen_fragment_to_dataCollectionFragment cannot be found from the current destination Destination(org.openforis.ground:id/submissionConfirmationDialogFragment) label=DataSubmittedConfirmationDialogFragment
       at androidx.navigation.NavController.navigate(NavController.kt:1691)
       at androidx.navigation.NavController.navigate(NavController.kt:1609)
       at androidx.navigation.NavController.navigate(NavController.kt:2169)
       at com.google.android.ground.MainActivity.onNavigate(MainActivity.kt:157)
       at com.google.android.ground.MainActivity.access$onNavigate(MainActivity.kt:56)
       at com.google.android.ground.MainActivity$onCreate$2$1.emit(MainActivity.kt:82)
       at com.google.android.ground.MainActivity$onCreate$2$1.emit(MainActivity.kt:82)
       at kotlinx.coroutines.flow.SharedFlowImpl.collect$suspendImpl(SharedFlow.kt:382)
       at kotlinx.coroutines.flow.SharedFlowImpl$collect$1.invokeSuspend(:15)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
       at kotlinx.coroutines.DispatchedTaskKt.resume(DispatchedTask.kt:235)
       at kotlinx.coroutines.DispatchedTaskKt.resumeUnconfined(DispatchedTask.kt:191)
       at kotlinx.coroutines.DispatchedTaskKt.dispatch(DispatchedTask.kt:163)
       at kotlinx.coroutines.CancellableContinuationImpl.dispatchResume(CancellableContinuationImpl.kt:474)
       at kotlinx.coroutines.CancellableContinuationImpl.resumeImpl(CancellableContinuationImpl.kt:508)
       at kotlinx.coroutines.CancellableContinuationImpl.resumeImpl$default(CancellableContinuationImpl.kt:497)
       at kotlinx.coroutines.CancellableContinuationImpl.resumeWith(CancellableContinuationImpl.kt:368)
       at kotlinx.coroutines.flow.SharedFlowImpl.tryEmit(SharedFlow.kt:399)
       at kotlinx.coroutines.flow.SharedFlowImpl.emit$suspendImpl(SharedFlow.kt:404)
       at kotlinx.coroutines.flow.SharedFlowImpl.emit()
       at kotlinx.coroutines.flow.SharedFlowImpl.collect$suspendImpl(SharedFlow.kt:382)
       at kotlinx.coroutines.flow.SharedFlowImpl$collect$1.invokeSuspend(:15)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
       at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
       at android.os.Handler.handleCallback(Handler.java:958)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loopOnce(Looper.java:230)
       at android.os.Looper.loop(Looper.java:319)
       at android.app.ActivityThread.main(ActivityThread.java:8919)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:578)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
        
```

```
Fatal Exception: java.lang.IllegalArgumentException: Navigation action/destination org.openforis.ground:id/showTermsOfService cannot be found from the current destination Destination(org.openforis.ground:id/terms_of_service_fragment) label=Terms of service class=com.google.android.ground.ui.tos.TermsOfServiceFragment
       at androidx.navigation.NavController.navigate(NavController.kt:1691)
       at androidx.navigation.NavController.navigate(NavController.kt:1609)
       at androidx.navigation.NavController.navigate(NavController.kt:2169)
       at com.google.android.ground.MainActivity.onNavigate(MainActivity.kt:157)
       at com.google.android.ground.MainActivity.access$onNavigate(MainActivity.kt:56)
       at com.google.android.ground.MainActivity$onCreate$2$1.emit(MainActivity.kt:82)
       at com.google.android.ground.MainActivity$onCreate$2$1.emit(MainActivity.kt:82)
       at kotlinx.coroutines.flow.SharedFlowImpl.collect$suspendImpl(SharedFlow.kt:382)
       at kotlinx.coroutines.flow.SharedFlowImpl$collect$1.invokeSuspend(:15)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
       at kotlinx.coroutines.DispatchedTaskKt.resume(DispatchedTask.kt:235)
       at kotlinx.coroutines.DispatchedTaskKt.resumeUnconfined(DispatchedTask.kt:191)
       at kotlinx.coroutines.DispatchedTaskKt.dispatch(DispatchedTask.kt:163)
       at kotlinx.coroutines.CancellableContinuationImpl.dispatchResume(CancellableContinuationImpl.kt:474)
       at kotlinx.coroutines.CancellableContinuationImpl.resumeImpl(CancellableContinuationImpl.kt:508)
       at kotlinx.coroutines.CancellableContinuationImpl.resumeImpl$default(CancellableContinuationImpl.kt:497)
       at kotlinx.coroutines.CancellableContinuationImpl.resumeWith(CancellableContinuationImpl.kt:368)
       at kotlinx.coroutines.flow.SharedFlowImpl.tryEmit(SharedFlow.kt:399)
       at kotlinx.coroutines.flow.SharedFlowImpl.emit$suspendImpl(SharedFlow.kt:404)
       at kotlinx.coroutines.flow.SharedFlowImpl.emit()
       at kotlinx.coroutines.flow.SharedFlowImpl.collect$suspendImpl(SharedFlow.kt:382)
       at kotlinx.coroutines.flow.SharedFlowImpl$collect$1.invokeSuspend(:15)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
       at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
       at android.os.Handler.handleCallback(Handler.java:942)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loopOnce(Looper.java:201)
       at android.os.Looper.loop(Looper.java:288)
       at android.app.ActivityThread.main(ActivityThread.java:8181)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:573)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1042)
        
```

```
Fatal Exception: java.lang.IllegalArgumentException: Navigation action/destination org.openforis.ground:id/showOfflineAreas cannot be found from the current destination Destination(org.openforis.ground:id/offline_areas_fragment) label=Offline map imagery class=com.google.android.ground.ui.offlineareas.OfflineAreasFragment
       at androidx.navigation.NavController.navigate(NavController.kt:1691)
       at androidx.navigation.NavController.navigate(NavController.kt:1609)
       at androidx.navigation.NavController.navigate(NavController.kt:2169)
       at com.google.android.ground.MainActivity.onNavigate(MainActivity.kt:157)
       at com.google.android.ground.MainActivity.access$onNavigate(MainActivity.kt:56)
       at com.google.android.ground.MainActivity$onCreate$2$1.emit(MainActivity.kt:82)
       at com.google.android.ground.MainActivity$onCreate$2$1.emit(MainActivity.kt:82)
       at kotlinx.coroutines.flow.SharedFlowImpl.collect$suspendImpl(SharedFlow.kt:382)
       at kotlinx.coroutines.flow.SharedFlowImpl$collect$1.invokeSuspend(:15)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
       at kotlinx.coroutines.DispatchedTaskKt.resume(DispatchedTask.kt:235)
       at kotlinx.coroutines.DispatchedTaskKt.resumeUnconfined(DispatchedTask.kt:191)
       at kotlinx.coroutines.DispatchedTaskKt.dispatch(DispatchedTask.kt:163)
       at kotlinx.coroutines.CancellableContinuationImpl.dispatchResume(CancellableContinuationImpl.kt:474)
       at kotlinx.coroutines.CancellableContinuationImpl.resumeImpl(CancellableContinuationImpl.kt:508)
       at kotlinx.coroutines.CancellableContinuationImpl.resumeImpl$default(CancellableContinuationImpl.kt:497)
       at kotlinx.coroutines.CancellableContinuationImpl.resumeWith(CancellableContinuationImpl.kt:368)
       at kotlinx.coroutines.flow.SharedFlowImpl.tryEmit(SharedFlow.kt:399)
       at kotlinx.coroutines.flow.SharedFlowImpl.emit$suspendImpl(SharedFlow.kt:404)
       at kotlinx.coroutines.flow.SharedFlowImpl.emit()
       at kotlinx.coroutines.flow.SharedFlowImpl.collect$suspendImpl(SharedFlow.kt:382)
       at kotlinx.coroutines.flow.SharedFlowImpl$collect$1.invokeSuspend(:15)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
       at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
       at android.os.Handler.handleCallback(Handler.java:958)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loopOnce(Looper.java:230)
       at android.os.Looper.loop(Looper.java:319)
       at android.app.ActivityThread.main(ActivityThread.java:8913)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:608)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
        
```

### PR DESCRIPTION
**Pull request by anandwana001**
Fixes #2776 

@shobhitagarwal1612  PTAL?

Removed Navigator from TermsOfServiceFragment, SurveySelectorFragment and OfflineAreasFragment

Part 2 - https://github.com/google/ground-android/pull/2896

### CODE REVIEW

**Comment by shobhitagarwal1612** [comment]
What would be the side-effect of this fix? The navigation won't happen so the screen won't be moved to the next destination? Can you please try and replicate one scenario forcefully to understand what the user will see?

**Comment by anandwana001** [comment]
The user will remain on the screen. However, this type of condition comes to the user in a worst case where the naviations instance are having are issue.

**Comment by shobhitagarwal1612** [comment]
I'm not sure if we should silently let the app fail. 

@gino-m Thoughts?

**Comment by shobhitagarwal1612** [comment]
@anandwana001 Can you look into options on how to resolve this problem?

**Comment by anandwana001** [comment]
@gino-m @shobhitagarwal1612 I did take a look at all the logs again: (updated issue https://github.com/google/ground-android/issues/2776) logs

1. submissionConfirmationDialogFragment - no more exists
2. showTermsOfService - the terms_of_service_fragment is trying to open itself again using the navigate() function
3. showOfflineAreas - the offline_areas_fragment is trying to open itself again using the navigate() function

Not reproable, but as per the understanding the `_navigateRequests` flow which is the source emitting the destinations is fixed in this PR, as earlier the scope was different.

**Comment by gino-m** [comment]
> Not reproable, but as per the understanding the `_navigateRequests` flow which is the source emitting the destinations is fixed in this PR, as earlier the scope was different.

In this PR, the lifecycle is still bound to the Activity, which is the wrong scope, since the fragment requesting the navigation may already be gone by the time the Activity receives the requests. 

Can we instead remove `Navigator` and related code in `MainActivity`, an instead opt for the simpler approach of ViewModel-scoped navigation as described in https://developer.android.com/guide/navigation?

**Review by gino-m — CHANGES_REQUESTED** [review]

File: ground/src/main/java/com/google/android/ground/ui/offlineareas/selector/OfflineAreaSelectorFragment.kt [hunk 1]
```diff
@@ -64,6 +67,22 @@ class OfflineAreaSelectorFragment : AbstractMapContainerFragment() {
     viewModel.isDownloadProgressVisible.observe(viewLifecycleOwner) {
       showDownloadProgressDialog(it)
     }
+
+    viewLifecycleOwner.lifecycleScope.launch {
+      viewLifecycleOwner.lifecycle.repeatOnLifecycle(Lifecycle.State.RESUMED) {
```
[the remark this answers is not in the document]
Comment (anandwana001) [reply ground/src/main/java/com/google/android/ground/ui/offlineareas/selector/OfflineAreaSelectorFragment.kt:72]: `viewLifecycleOwner.lifecycleScope.launch {}` 
 Here, the `lifecycleScope` is tied to the view of the fragment
 
 `lifecycleScope.launch {}` 
 Here, the `lifecycleScope` is tied to the fragment itself
 
 https://developer.android.com/guide/fragments/lifecycle

[the remark this answers is not in the document]
Comment (anandwana001) [reply ground/src/main/java/com/google/android/ground/ui/offlineareas/selector/OfflineAreaSelectorFragment.kt]: updated

**Review by gino-m — APPROVED** [review]
In many cases case `UiState` isn't actually modeling states, but actually actions/events. Can we rename those to `NavEvent` in a fast-follow?

Also, in cases `UiState` classes are used in a `SharedStateFlow`, could we please switch those over the a `SharedFlow` if it makes sense?

Filed https://github.com/google/ground-platform/issues/2105 to track.

**Pull request merged**